src/Storages/MergeTree/GinIndexStore.cpp

```diff
@@ -171,48 +165,19 @@ bool GinIndexStore::exists() const
     return storage->existsFile(segment_id_file_name);
 }
 
-UInt32 GinIndexStore::getNextSegmentIDRange(const String & file_name, size_t n)
+UInt32 GinIndexStore::getNextSegmentIDRange(size_t n)
 {
     std::lock_guard guard(mutex);
 
-    /// When the method is called for the first time, the file doesn't exist yet, need to create it and write segment ID 1.
-    if (!storage->existsFile(file_name))
-    {
-        /// Create file
-        std::unique_ptr<DB::WriteBufferFromFileBase> ostr = this->data_part_storage_builder->writeFile(file_name, 8, {});
-
-        /// Write version
-        writeChar(static_cast<char>(CURRENT_GIN_FILE_FORMAT_VERSION), *ostr);
-
-        /// Write segment ID 1
-        writeVarUInt(1, *ostr);
-        ostr->sync();
-        ostr->finalize();
-    }
-
-    /// Read id in file
-    UInt32 result = 0;
-    {
-        std::unique_ptr<DB::ReadBufferFromFileBase> istr = this->storage->readFile(file_name, {}, std::nullopt, std::nullopt);
-
-        /// Skip version
-        istr->seek(1, SEEK_SET);
-
-        readVarUInt(result, *istr);
-    }
-
-    /// Save result + n
-    {
-        std::unique_ptr<DB::WriteBufferFromFileBase> ostr = this->data_part_storage_builder->writeFile(file_name, 8, {});
-
-        /// Write version
-        writeChar(static_cast<char>(CURRENT_GIN_FILE_FORMAT_VERSION), *ostr);
+    /// We cannot directly use file system to read and update the segment id,
+    /// because packed storage do not support reading value from the uncommitted file.
```

Comment:
For someone who doesn't know the previous implementation, the comment may be confusing. I would say the way it is implemented now is how one would do it intuitively. Therefore, suggest to remove the comment (also in l. 174).